src/Util/File.cc

```diff
@@ -67,3 +67,89 @@ fs::path SearchPath(std::string filename) {
     }
     return fs::path();
 }
+
+CARTA::FileType GuessImageType(const std::string& path_string, bool check_content) {
+    if (check_content) {
+        // Guess file type by magic number
+        auto magic_number = GetMagicNumber(path_string);
+        switch (magic_number) {
+            case FITS_MAGIC_NUMBER:
+                return CARTA::FITS;
+            case HDF5_MAGIC_NUMBER:
+                return CARTA::HDF5;
+            case GZ_MAGIC_NUMBER:
+                fs::path gz_path(path_string);
+                std::string extension = gz_path.stem().extension().string();
+                return HasSuffix(extension, ".fits") ? CARTA::FITS : CARTA::UNKNOWN;
+        }
+    } else {
+        // Guess file type by extension
+        fs::path path(path_string);
+        auto filename = path.filename().string();
+
+        if (HasSuffix(filename, ".fits") || HasSuffix(filename, ".fz") || HasSuffix(filename, ".fits.gz")) {
```

Comment:
These are all case-sensitive checks, because the flag defaults to false.

(I would consider making it default to true, although that may be too misleading.)

Reply:
the flag is `case_sensitive`, so setting it to false means the checks are all case-insensitive.

Reply:
Ohhh, I see now. :)